### ISSUE TITLE
Physical keyboard focus does not work with TextInput using Tab/Shift+Tab on Android

### ISSUE THREAD

**Issue by kriti18singh** [microsoft/react-native-macos#782]
### Environment
1. `react-native -v`:
3. `npm ls react-native-macos`:
4. `node -v`: v12.18.3
5. `npm -v`: 6.14.6
6. `yarn --version`: 1.22.5
7. `xcodebuild -version`: Xcode 12.4 Build version 12D4e

### Issue
Using a physical keyboard (bluetooth or usb) to move the focus using Tab/Shift + Tab keys  into a TextInput does not work on Android. Tested this with a real Android device.
### Steps to Reproduce

The layout looks like this  : 
![temp](https://user-images.githubusercontent.com/18608953/119641974-dcb10700-be37-11eb-86e9-e43a40e6df2a.jpg)

1. Connect a hardware keyboard to your Android phone using bluetooth or Usb.
2. Use Tab key to go to the text input -> focus does not move to the TextInput and is stuck at the last focusable element, button 1 in this case.
3. Use Shit+Tab key to move focus from the next focusable element( button 2), back to the TextInput. The focus stays with button 2 and does not move into the text input component.

### Expected Behavior

The tab and shit+tab keys should work to move the focus into the TextInput from previous focusable elements.
### Actual Behavior

The keyboard focus is trapped with the focusable elements and does not move into the TextInput with Tab or Shift Tab.
### Reproducible Demo
This is easily reproducible with any TextInput component while using hardware keyboard.

### Additional context

If I click inside the TextInput (giving it the focus), and then Tab to go to Button 2, the focus moves. However, it does not work from the Button to the text input.

**Comment by chrisglein** [comment]
@kriti18singh Just to confirm, does this occur with RN Android without using the macOS fork? As in do we know this is a result of forked behavior that is affecting Android? If this repros on RN core we'll want to move this issue to their repo.

**Comment by chrisglein** [comment]
> @chrisglein Chris Glein FTE I can repro this on both the open source version as well as the fork.

Thanks for the clarification, @kriti18singh! I can't transfer issues across organizations. Can you open a fresh issue on the RN core repo here https://github.com/facebook/react-native/issues/new/choose?
